### DISCUSSION TITLE
How to do simulation of flexible object in pybullet

### DISCUSSION THREAD

**Discussion by houliwei** [bulletphysics/bullet3#3593] [General]

**Comment by houliwei** [comment]
Can pybullet model a flexible object with certain stiffness ？

**Comment by adverley** [comment]
Bullet contains soft body dynamics to a certain degree. To examine the options, look at [this](https://www.youtube.com/watch?v=9sh-D0nejUY) youtube video or build Bullet with the soft body option, run the example browser and look at the soft body examples. 

PyBullet has some bindings to the soft body part of the engine, see the [examples](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/load_soft_body.py). Do mind that this is very experimental, old code and not as stable as the rigid body simulation of the physics engine.

**Comment by erwincoumans** [comment]
Some people used the deformable simulation in PyBullet, see https://sites.google.com/view/sim-to-real-deformable
We don't have many resources for this, so we focus on the rigid/multibody/collision detection parts.

**Comment by erwincoumans** [comment]
You can also use createConstraint with a fixed constraint using limited force to mimic soft constraints.
See the constraint.zip example here: https://github.com/bulletphysics/bullet3/issues/2100#issuecomment-461621661

**Comment by DanielTakeshi** [comment]
@houliwei  We have been able to use deformables in PyBullet in this recent work: https://berkeleyautomation.github.io/bags/

**Comment by zhiwuz** [comment]
Hi everyone. I came across this discussion. I recently developed simulation of flexible objects with PyBullet by modeling it as rigid links connected by joints (torque controlled). The torque is determined by physics of bending stiffness. So it can model flexible objects with a certain bending stiffness, while assuming it can not be extended in length. The paper is here:[https://ieeexplore.ieee.org/document/9811927]( https://ieeexplore.ieee.org/document/9811927 )
